datasources.go

```diff
@@ -36,6 +42,16 @@ func (api API) CreateDataSource(name string) (*DataSource, error) {
 	return ds, err
 }
 
+// CreateDataSource2 creates an API Data Source in ChartMogul.
+// * Allows other parameters than just the name.
+//
+// See https://dev.chartmogul.com/v1.0/reference#data-sources
+func (api API) CreateDataSource2(dataSource *DataSource) (*DataSource, error) {
```

Comment:
@pkopac not sure if this makes sense in Go, but an alternative in some languages is to make the new functions part of a new namespace/package. The rationale is that having a weird package name is less intrusive (since it doesn't appear in each function call) and conveys that both versions of the function serve the same purpose.

Also, if the reason for the previous versions to exist is for keeping backward compatibility, would it make sense to mark them as deprecated in some way? Even if there isn't a good way to do so, maybe just adding a code comment?

Anyway, I think it is a good idea to not introduce unnecessary breakage, especially when there are people depending on this. Kudos for that! 👍